src/core/undefined_struct.rs

```diff
@@ -205,15 +204,28 @@ impl<'a> UndefinedStructTable {
             .and_then(|undefined_struct| Some(T::new(&undefined_struct)))
     }
 
+    /// Finds the first occurance of the structure that satisfies a predicate.
+    pub fn find_defined_struct<T, P>(&'a self, mut predicate: P) -> Option<T>
```

Comment:
Would it be possible to have an example or a unit test for this new pub fn? Not sure how a `predicate` should like.

Reply:
Yes in fact I realized after I shutdown the computer last night I wrote it incorrectly, FnMut needs to take (&T) as a parameter instead so that the person calling work with the implementation of T, not the undefined struct behind it.  A test would surely have demonstrated this design error.

Reply:
Fixed